central/cluster/datastore/datastore_impl.go

```diff
@@ -152,6 +155,24 @@ func (ds *datastoreImpl) UpdateClusterStatus(ctx context.Context, id string, sta
 	return ds.clusterStorage.Upsert(ctx, cluster)
 }
 
+func (ds *datastoreImpl) removeImageIntegrationByCluster(ctx context.Context, cluster *storage.Cluster) {
+
+	q := pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.ClusterID, cluster.GetId()).ProtoQuery()
+	log.Info(">>>> Image Integrations: " + q.String() + " " + q.GetBaseQuery().String())
+	imageIntegrations, err := ds.imageIntegrationDataStore.Search(ctx, q)
+	log.Infof(">>>>Image Integrations list size is %d", len(imageIntegrations))
```

Comment:
If the two log lines here are for local debugging purpose, make sure to remove them before merging.